### PR TITLE
Automate creation of custom spack-stack Dockerfile

### PR DESCRIPTION
**Pull request by christopherwharrop-noaa**
This PR removes the customized Dockerfile for spack-stack and replaces it with a script that generates that custom Dockerfile from spack-stack's official repository and tools.  The CI was updated to run this script to create the container Dockerfile before attempting to build it.

### CODE REVIEW

**Comment by christopherwharrop-noaa** [comment]
@NaureenBharwaniNOAA - This should make creation of the spack-stack base layer container more reliable.

**Comment by NaureenBharwaniNOAA** [comment]
Is this PR ready for review? I'll take a look at shortly if so.

**Comment by christopherwharrop-noaa** [comment]
Yes.  Please review it when you have time.

File: docker/test/src/parsl_flux_mpi_hello.py [hunk 1]
```diff
@@ -63,48 +72,63 @@ def pmi_barrier(parsl_resource_specification={}):
...
+
+# run the mpi app with chiltepin stack
+hello = mpi_hello(dirpath=shared_dir,
+                  stdout=os.path.join(shared_dir, "parsl_flux_mpi_hello_run.out"),
+                  stderr=os.path.join(shared_dir, "parsl_flux_mpi_hello_run.err",),
+                  stack=chiltepin,
+                  parsl_resource_specification={"num_tasks": 6, "num_nodes": 3})
+
+# Wait for the MPI app with chiltepin stack to finish
+hello.result()
```
Comment (NaureenBharwaniNOAA) [inline docker/test/src/parsl_flux_mpi_hello.py:115]: This looks like a future similar to what they talk about in Parsl and I just wanted to confirm that it is a future. 
The comment suggests it is waiting for the app to finish running an return the result.

Comment (christopherwharrop-noaa) [reply docker/test/src/parsl_flux_mpi_hello.py:115]: Yes.  Calling `mpi_hello` returns an App Future which gets assigned to `hello`.  Calling `result()` on the App Future object causes the program to block while waiting for the future to complete.

**Pull request merged**